### ISSUE TITLE
DAWNBench: An End-to-End Deep Learning Benchmark and Competition

### ISSUE THREAD

**Issue by icoxfog417** [arXivTimes/arXivTimes#648]
## 一言でいうと

ある精度を達成するのに、どれくらい時間/コストがかかるかを検証した研究。このGPU、このライブラリ(TensorFlowなど)の実装でこれくらい時間がかかる、というように、モデルだけでなくそれを走らせるハードまで含めたEnd to Endのコストを検証している。

### 著者/所属機関

Cody Coleman, Deepak Narayanan, Daniel Kang, Tian Zhao, Jian Zhang, Luigi Nardi, Peter Bailis, Kunle Olukotun, Chris Ré, Matei Zaharia

* Stanford DAWN Project

### 投稿日付(yyyy/MM/dd)

NIPS 2017

## 結果

1. minibatchが収束と計算速度に与える影響。
小さいほど収束が良く、大きいほど計算速度が速いというトレードオフ(CIFAR10/ResNet56)。

![image](https://user-images.githubusercontent.com/544269/36288377-b1572794-12fd-11e8-8043-eeadbc3e74f6.png)

2.optimizerの影響

収束のSGDと速度のAdam(CIFAR10/ResNet56)

![image](https://user-images.githubusercontent.com/544269/36288461-0b819a74-12fe-11e8-981a-18482f6c617c.png)

CIFAR10/ResNet110を1 Nvidia K80 GPU, minibatch=128, initial learning rate=0.1(SGD?)で学習させたのをBaseとして、SD(stochastic depth)、4GPU分散、minibatch size (BS) = 512を加えていったものの検証(左から徐々に加えていき、中央から右は徐々に抜いていっている)。minibatchのサイズをでかくするのは精度面で大きなトレードオフがあることがわかる。

![image](https://user-images.githubusercontent.com/544269/36288484-34b5738e-12fe-11e8-8beb-b66b3a292a8c.png)

など

3.ハードウェアと実装ライブラリによる影響(TensorFlow vs PyTorchをいろんなプラットフォームで)

CPUではTensorFlowの方が早いが、GPUではPyTorchの方が早い(というかTensorFlowがかなり遅い)。
なお、公式サイトではMXNetなど他のフレームワークの結果も掲載されている。

## コメント

公式サイトはこちら
https://dawn.cs.stanford.edu/benchmark/
